src/main/java/org/javers/organization/structure/ui/OrganizationTree.java

```diff
@@ -0,0 +1,194 @@
+            public void valueChange(Property.ValueChangeEvent event) {
+                selected = (Hierarchy) comboBox.getValue();
+                container.removeAllItems();
+                populateContainer(selected);
+            }
+        });
+
+        hierarchies.forEach(i -> comboBox.setItemCaption(i, i.getHierarchyName()));
+        container.addContainerProperty("login", String.class, null);
+        tree.setContainerDataSource(container);
+        tree.setItemCaptionPropertyId("login");
+        tree.setDragMode(Tree.TreeDragMode.NODE);
+        tree.setDropHandler(new TreeSortDropHandler(tree));
+
+        tree.addValueChangeListener(new Property.ValueChangeListener() {
+            @Override
+            public void valueChange(Property.ValueChangeEvent event) {
+                Employee e = (Employee) tree.getValue();
+                if (e!=null) {
+                controller.employeeSelected(e);
+
+                }
+
+            }
+        });
+
+        populateContainer(selected);
+    }
+
+    private void populateContainer(Hierarchy first) {
+        Employee root = first.getRoot();
+
+        Item item = container.addItem(root);
+        item.getItemProperty("login").setValue(root.getLogin());
+        addSubordinates(root);
+        tree.rootItemIds().forEach(i -> tree.expandItemsRecursively(i));
+    }
+
+    private void addSubordinates(Employee root) {
+        for (Employee e : root.getAllEmployees()) {
+            Item item = container.addItem(e);
+            item.getItemProperty("login").setValue(e.getLogin());
+            container.setParent(e, root);
+            addSubordinates(e);
+        }
+    }
+
+    //MAGIC
+    private static class TreeSortDropHandler implements DropHandler {
+        private final Tree tree;
+
+        public TreeSortDropHandler(final Tree tree) {
+            this.tree = tree;
+        }
+
+        @Override
+        public AcceptCriterion getAcceptCriterion() {
+            // Alternatively, could use the following criteria to eliminate some
+            // checks in drop():
+            // new And(IsDataBound.get(), new DragSourceIs(tree));
+            return AcceptAll.get();
+        }
+
+        @Override
+        public void drop(final DragAndDropEvent dropEvent) {
+            final Transferable t = dropEvent.getTransferable();
+            if (t.getSourceComponent() != tree || !(t instanceof DataBoundTransferable)) {
+                return;
+            }
+            final Tree.TreeTargetDetails dropData = ((Tree.TreeTargetDetails) dropEvent.getTargetDetails());
+            final Object sourceItemId = ((DataBoundTransferable) t).getItemId();
+            final Object targetItemId = dropData.getItemIdOver();
+            final VerticalDropLocation location = dropData.getDropLocation();
+
+
+            moveNode(sourceItemId, targetItemId, location);
+        }
+
+        /**
+         * Move a node within a tree onto, above or below another node depending
```

Comment:
to jest skopiowane z netu czy masz taki skill w Vadinie :) ?
jeśli skopiowane to podajmy w javadocu źródło (url)